src/frequenz/dispatch/_actors_service.py

```diff
@@ -0,0 +1,225 @@
+
+        def _update_dispatch_information(self, dispatch_update: DispatchUpdate) -> None:
+            print("Received update:", dispatch_update)
+            self._dry_run = dispatch_update.dry_run
+            self._options = dispatch_update.options
+            match dispatch_update.components:
+                case []:
+                    print("Dispatch: Using all components")
+                case list() as ids if isinstance(ids[0], int):
+                    component_ids = ids
+                case [ComponentCategory.BATTERY, *_]:
+                    component_category = ComponentCategory.BATTERY
+                case unsupported:
+                    print(
+                        "Dispatch: Requested an unsupported selector %r, "
+                        "but only component IDs or category BATTERY are supported.",
+                        unsupported,
+                    )
+
+    async def main():
+        url = os.getenv("DISPATCH_API_URL", "grpc://fz-0004.frequenz.io:50051")
+        key  = os.getenv("DISPATCH_API_KEY", "some-key")
+
+        microgrid_id = 1
+
+        dispatcher = Dispatcher(
+            microgrid_id=microgrid_id,
+            server_url=url,
+            key=key
+        )
+        dispatcher.start()
+
+        status_receiver = dispatcher.new_running_state_event_receiver("EXAMPLE_TYPE")
+
+        def actor_factory(initial_dispatch, dispatch_receiver):
+            return MyActor.new_with_dispatch(initial_dispatch, dispatch_receiver)
+
+        managing_actor = DispatchManagingActor(
+            actor_factory=actor_factory,
+            running_status_receiver=status_receiver,
+        )
+
+        await run(managing_actor)
+    ```
+    """
+
+    def __init__(
+        self,
+        actor_factory: Callable[[DispatchUpdate, Receiver[DispatchUpdate]], Actor],
+        running_status_receiver: Receiver[Dispatch],
+    ) -> None:
+        """Initialize the dispatch handler.
+
+        Args:
+            actor_factory: A callable that creates an actor with some initial dispatch
+                information.
+            running_status_receiver: The receiver for dispatch running status changes.
+        """
+        super().__init__()
+        self._dispatch_rx = running_status_receiver
+        self._actor_factory = actor_factory
+        self._actor: Actor | None = None
+        self._updates_channel = Broadcast[DispatchUpdate](
+            name="dispatch_updates_channel", resend_latest=True
+        )
+        self._updates_sender = self._updates_channel.new_sender()
+
+    def start(self) -> None:
+        """Start the background service."""
+        self._tasks.add(asyncio.create_task(self._run()))
+
+    def new_receiver(self) -> Receiver[DispatchUpdate]:
+        """Create a new receiver for dispatch updates.
+
+        Returns:
+            A new receiver for dispatch updates.
+        """
+        return self._updates_channel.new_receiver()
```

Comment:
Do you really need to expose this? I wouldn't add a public method for this if it is basically only used internally.